cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +89,410 @@ def unset_default_org(self):
+            extra=[name, self],
+            context=f"org config ({name})",
+        )
+        if self.orgs[name].global_org:
+            org.global_org = True
+        return org
+
+    def _remove_org(self, name, global_org):
+        scope = self.global_config_dir if global_org else self.project_local_dir
+        org_path = Path(f"{scope}/{name}.org")
+        if not org_path.exists():
+            if not global_org:
+                raise OrgNotFound(
+                    f"Could not find org named {name} to delete.  Deleting in project org mode.  Is {name} a global org?"
+                )
+            raise OrgNotFound(
+                f"Could not find org named {name} to delete.  Deleting in global org mode.  Is {name} a project org instead of a global org?"
+            )
+
+        org_path.unlink()
+        del self.orgs[name]
+
+    def _raise_org_not_found(self, name):
+        raise OrgNotFound(
+            f"Org information could not be found. Expected to find encrypted file at {self.project_local_dir}/{name}.org"
+        )
+
+    #######################################
+    #              Services               #
+    #######################################
+
+    def set_default_service(
+        self, service_type: str, alias: str, project: bool = False
+    ) -> None:
+        """Public API for setting a default service e.g. `cci service default`"""
+        if service_type not in self.project_config.services:
+            raise ServiceNotConfigured(f"No such service type: {service_type}")
+        elif alias not in self.services[service_type]:
+            raise ServiceNotConfigured(
+                f"No service of type {service_type} configured with name: {alias}"
+            )
+        elif alias == self._default_services[service_type]:
+            return  # already the default
+
+        self._default_services[service_type] = alias
+        self._save_default_service(service_type, alias, project=project)
+
+    def rename_service(
+        self, service_type: str, current_alias: str, new_alias: str
+    ) -> None:
+        """Public API for renaming a service
+
+        @param service_type type of service being renamed
+        @param current_alias the current alias of the service
+        @param new_alias the new alias for the service
+        @throws: ServiceNotValid if no services of the given type are configured,
+        or if no service of the given type has the current_alias
+        """
+
+        if service_type not in self.services:
+            raise ServiceNotConfigured(
+                f"No services of type {service_type} are currently configured"
+            )
+        elif current_alias not in self.services[service_type]:
+            raise ServiceNotConfigured(
+                f"No service of type {service_type} configrued with the name: {current_alias}"
+            )
+
+        self.services[service_type][new_alias] = self.services[service_type][
+            current_alias
+        ]
```

Comment:
Don't we need a check to make sure the new_alias doesn't already exist?

Reply:
good catch